src/Microsoft.AspNet.Mvc.TagHelpers/HiddenFormTagHelper.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting all form elements to generate hidden tag for checkboxes.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Advanced)]
+    [HtmlTargetElement("form")]
+    public class HiddenFormTagHelper : TagHelper
```

Comment:
Need to be more general w/ these descriptions and names. Check boxes are one use of the infrastructure this helper provides. Perhaps `... to generate deferred content before the &lt;/form&gt; end tag.` and `RenderBeforeFormEndTagTagHelper`?